apps/frontend/src/hooks/aave/useAaveRates.tsx

```diff
@@ -0,0 +1,300 @@
+            internalType: 'uint256',
+            name: 'totalVariableDebt',
+            type: 'uint256',
+          },
+          {
+            internalType: 'uint256',
+            name: 'reserveFactor',
+            type: 'uint256',
+          },
+          {
+            internalType: 'address',
+            name: 'reserve',
+            type: 'address',
+          },
+          {
+            internalType: 'address',
+            name: 'aToken',
+            type: 'address',
+          },
+        ],
+        internalType: 'struct DataTypes.CalculateInterestRatesParams',
+        name: 'params',
+        type: 'tuple',
+      },
+    ],
+    name: 'calculateInterestRates',
+    outputs: [
+      {
+        internalType: 'uint256',
+        name: '',
+        type: 'uint256',
+      },
+      {
+        internalType: 'uint256',
+        name: '',
+        type: 'uint256',
+      },
+      {
+        internalType: 'uint256',
+        name: '',
+        type: 'uint256',
+      },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  },
+  {
+    inputs: [],
+    name: 'getBaseVariableBorrowRate',
+    outputs: [
+      {
+        internalType: 'uint256',
+        name: '',
+        type: 'uint256',
+      },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  },
+  {
+    inputs: [],
+    name: 'getMaxVariableBorrowRate',
+    outputs: [
+      {
+        internalType: 'uint256',
+        name: '',
+        type: 'uint256',
+      },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  },
+  {
+    inputs: [],
+    name: 'getVariableRateSlope1',
+    outputs: [
+      {
+        internalType: 'uint256',
+        name: '',
+        type: 'uint256',
+      },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  },
+  {
+    inputs: [],
+    name: 'getVariableRateSlope2',
+    outputs: [
+      {
+        internalType: 'uint256',
+        name: '',
+        type: 'uint256',
+      },
+    ],
+    stateMutability: 'view',
+    type: 'function',
+  },
+];
+
+export interface RatesDataResult {
+  currentUsageRatio: string;
+  optimalUsageRatio: string;
+  baseVariableBorrowRate: string;
+  variableRateSlope1: string;
+  variableRateSlope2: string;
+  underlyingAsset: string;
+  name: string;
+  symbol: string;
+  decimals: string;
+}
+
+const calculateUtilizationRate = (
+  decimals: number,
+  totalDebt: string,
+  availableLiquidity: string,
+): bigint => {
+  // Create BigNumber instances
+  const totalBorrow = BigInt(utils.parseUnits(totalDebt, decimals).toString());
+  const totalSupply = BigInt(availableLiquidity) + totalBorrow;
+  // Perform division
+  return (totalBorrow * BigInt(10 ** 18)) / totalSupply;
+};
+
+export const useAaveInterestRatesData = (): {
+  data: RatesDataResult | null;
+  error: string | null;
+} => {
+  const [data, setData] = useState<RatesDataResult | null>(null);
+  const [error, setError] = useState<string | null>(null);
+
+  const { provider } = useAccount();
```

Comment:
Should use `getProvider(BOB_CHAIN_ID)` to avoid depending on user connecting wallet to see graphs